vengine/vengine/resource_management/loaders/TextureLoader.cpp

```diff
@@ -28,31 +28,37 @@ void TextureLoader::init(
     this->resourceMan = resourceMan;
 }
 
-std::vector<std::string> TextureLoader::assimpGetTextures(const aiScene *scene) 
+void TextureLoader::assimpGetTextures(const aiScene *scene, std::vector<std::string> &diffuseTextures, std::vector<std::string> &emissiveTextures) 
 {
-    std::vector<std::string> textureList(scene->mNumMaterials);
+    diffuseTextures = std::vector<std::string>(scene->mNumMaterials);
```

Comment:
Please use "diffuseTextures.resize(scene->mNumMaterials)" instead of assigning. Same with emissiveTextures.